packages/main/src/components/AnalyticalTable/hooks/useAutoResize.tsx

```diff
@@ -0,0 +1,71 @@
+import { enrichEventWithDetails } from '@ui5/webcomponents-react-base';
+import { DEFAULT_COLUMN_WIDTH } from '../defaults/Column/index.js';
+import type { ReactTableHooks } from '../types/index.js';
+import { CELL_PADDING_PX } from './useDynamicColumnWidths.js';
+
+function setResizerProps(...params) {
+  const [props, { instance, header }] = params;
+  const { dispatch, virtualRowsRange, rows, webComponentsReactProperties, state } = instance;
+  const { uniqueId, onAutoResize } = webComponentsReactProperties;
+  const { autoResizable, id: accessor } = header;
+
+  if (!document || !autoResizable || !rows.length || !virtualRowsRange) {
+    return props;
+  }
+
+  return {
+    ...props,
+    onDoubleClick: (e) => {
+      let largest = getMeasureMax(accessor, uniqueId, virtualRowsRange, state.isRtl);
+      largest = largest > DEFAULT_COLUMN_WIDTH ? largest : DEFAULT_COLUMN_WIDTH;
+      onAutoResize(
+        enrichEventWithDetails(e, {
+          accessor,
+          width: largest
+        })
+      );
+      if (e.defaultPrevented) {
+        return;
+      }
+      dispatch({
+        type: 'AUTO_RESIZE',
+        payload: { [accessor]: largest }
+      });
+    }
+  };
+}
+
+function getMeasureMax(accessor, uniqueId, virtualRowsRange, isRtl) {
+  let maxWidth = 0;
+  const firstRowQuery = `[data-component-name="AnalyticalTableBodyScrollableContainer"][data-react-id="${uniqueId}"] [data-virtual-row-index="${virtualRowsRange.startIndex}"]`;
+  const rowsDOM = document.querySelectorAll(`${firstRowQuery}, ${firstRowQuery} ~ [data-virtual-row-index]`);
+  const start = virtualRowsRange.startIndex;
+  const end = virtualRowsRange.endIndex;
+
+  for (let i = 0; i <= end - start; i++) {
```

Comment:
The loop only measures rows that are currently visible, if the `querySelector` returns `undefined` it jumps to the next index.